cpp/include/cudf/io/json.hpp

```diff
@@ -668,7 +700,10 @@ class json_reader_options_builder {
    */
   json_reader_options_builder& delimiter(char delimiter)
   {
-    options.set_delimiter(delimiter);
+    if (json_reader_options::can_be_delimiter(delimiter))
+      options.set_delimiter(delimiter);
+    else
+      CUDF_FAIL("Unsupported delimiter character.", std::invalid_argument);
```

Comment:
`options` handles the error check.
```suggestion
    options.set_delimiter(delimiter);
```